src/domain/accounts/domain/src/sqlx/account_id_sqlx.rs

```diff
@@ -0,0 +1,46 @@
+// Copyright Kamu Data, Inc. and contributors. All rights reserved.
+//
+// Use of this software is governed by the Business Source License
+// included in the LICENSE file.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0.
+
+use opendatafabric::AccountID;
+
+/////////////////////////////////////////////////////////////////////////////////////////
+
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub struct AccountIdSqlx(AccountID);
```

Comment:
Curious why do this as newtype instead of implementing sqlx decode trait directly on `AccountID`?

If you are worried about coupling `AccountID` to `sqlx` - we could always put the trait impls under a feature flag.

Reply:
Plz tell me more about "feature flag". What I hated previously was a need to link `sqlx` crate in every using crate, which is logically very far from database.

Reply:
Here's an [example](https://github.com/kamu-data/kamu-cli/blob/755ec857fb77f420069b4b3e84dd65fea396ae48/src/domain/opendatafabric/src/dtos/dtos_extra.rs#L288) that enables some methods only if `arrow` feature is turned on.

Feature is declared [here](https://github.com/kamu-data/kamu-cli/blob/755ec857fb77f420069b4b3e84dd65fea396ae48/src/domain/opendatafabric/Cargo.toml#L26) and via `dep:arrow` it pulls in `arrow` as an [optional dependency](https://github.com/kamu-data/kamu-cli/blob/755ec857fb77f420069b4b3e84dd65fea396ae48/src/domain/opendatafabric/Cargo.toml#L62).

Reply:
Looks too rusty, but works.